test/instances/chainlink/ChainlinkTest.t.sol

```diff
@@ -0,0 +1,691 @@
+
+        // Ensure that the shares balances were updated correctly.
+        assertApproxEqAbs(
+            hyperdriveSharesAfter,
+            hyperdriveBalancesBefore.sharesBalance +
+                hyperdrive.convertToShares(amountPaid),
+            1
+        );
+        assertApproxEqAbs(
+            traderSharesAfter,
+            traderBalancesBefore.sharesBalance -
+                hyperdrive.convertToShares(amountPaid),
+            1
+        );
+    }
+
+    /// @dev Verifies that withdrawal accounting is correct when closing positions.
+    function verifyWithdrawal(
+        address trader,
+        uint256 baseProceeds,
+        bool asBase,
+        uint256 totalBaseBefore,
+        uint256 totalSharesBefore,
+        AccountBalances memory traderBalancesBefore,
+        AccountBalances memory hyperdriveBalancesBefore
+    ) internal view override {
+        // Base withdrawals are not supported for this instance.
+        if (asBase) {
+            revert IHyperdrive.UnsupportedToken();
+        }
+
+        // Ensure that the total supplies didn't change.
+        (uint256 totalBaseAfter, uint256 totalSharesAfter) = getSupply();
+        assertEq(totalBaseAfter, totalBaseBefore);
+        assertEq(totalSharesAfter, totalSharesBefore);
+
+        // Ensure that the ETH balances didn't change.
+        assertEq(
+            address(hyperdrive).balance,
+            hyperdriveBalancesBefore.ETHBalance
+        );
+        assertEq(bob.balance, traderBalancesBefore.ETHBalance);
+
+        // Ensure that the base balances do not change.
+        (
+            uint256 hyperdriveBaseAfter,
+            uint256 hyperdriveSharesAfter
+        ) = getTokenBalances(address(hyperdrive));
+        (uint256 traderBaseAfter, uint256 traderSharesAfter) = getTokenBalances(
+            trader
+        );
+        assertEq(hyperdriveBaseAfter, hyperdriveBalancesBefore.baseBalance);
+        assertEq(traderBaseAfter, traderBalancesBefore.baseBalance);
+
+        // Ensure that the shares balances were updated correctly.
+        assertApproxEqAbs(
+            hyperdriveSharesAfter,
+            hyperdriveBalancesBefore.sharesBalance -
+                hyperdrive.convertToShares(baseProceeds),
+            1
+        );
+        assertApproxEqAbs(
+            traderSharesAfter,
+            traderBalancesBefore.sharesBalance +
+                hyperdrive.convertToShares(baseProceeds),
+            1
+        );
+    }
+
+    /// Getters ///
+
+    function test_getters() external view {
+        assertEq(
+            address(IChainlinkHyperdrive(address(hyperdrive)).aggregator()),
+            address(CHAINLINK_AGGREGATOR_PROXY)
+        );
+        (, uint256 totalShares) = getTokenBalances(address(hyperdrive));
+        assertEq(hyperdrive.totalShares(), totalShares);
+    }
+
+    /// LP ///
+
+    function test_round_trip_lp_instantaneous(uint256 _contribution) external {
```

Comment:
would like to see high-level summary comments on what each of these tests are doing but can totally understand if that's not something we want to tackle rn

Reply:
I'll do this when I clean up the integration test suite.